src/main/java/com/conveyal/gtfs/validator/FlexValidator.java

```diff
@@ -0,0 +1,358 @@
+package com.conveyal.gtfs.validator;
+
+import com.conveyal.gtfs.error.NewGTFSError;
+import com.conveyal.gtfs.error.NewGTFSErrorType;
+import com.conveyal.gtfs.error.SQLErrorStorage;
+import com.conveyal.gtfs.loader.Feed;
+import com.conveyal.gtfs.model.BookingRule;
+import com.conveyal.gtfs.model.FareRule;
+import com.conveyal.gtfs.model.Location;
+import com.conveyal.gtfs.model.LocationGroup;
+import com.conveyal.gtfs.model.Stop;
+import com.conveyal.gtfs.model.StopTime;
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.conveyal.gtfs.model.Entity.DOUBLE_MISSING;
+import static com.conveyal.gtfs.model.Entity.INT_MISSING;
+import static com.conveyal.gtfs.util.GeoJsonUtil.GEOMETRY_TYPE_POLYGON;
+
+/**
+ * Spec validation checks for flex additions as defined here:
+ * https://github.com/MobilityData/gtfs-flex/blob/master/spec/reference.md
+ *
+ * Number of checks:
+ *
+ * Location group: 2
+ * Location: 2
+ * Stop times: 14
+ * Booking rules: 10
+ */
+public class FlexValidator extends FeedValidator {
+
+    // Store all validation errors here so that they can be checked as part of testing.
+    public final List<NewGTFSError> errors = new ArrayList<>();
+
+    /**
+     * Constructor for testing.
+     */
+    public FlexValidator() {
+        this(null, null);
+    }
+
+    public FlexValidator(Feed feed, SQLErrorStorage errorStorage) {
+        super(feed, errorStorage);
+    }
+
+    @Override
+    public void validate() {
+        feed.locationGroups.forEach(locationGroup ->
```

Comment:
There needs to be a check on whether a feed contains flex stuff, or that the corresponding tables are present, before this is allowed to proceed. When loading a large set of feeds without flex data, some of this code is repeatedly trying to hit the locations table and as a result takes very long to complete (or goes in an infinite loop?) while consuming a lot of CPU power. Running this locally, I had to kill the datatools-server process after a few minutes.

Reply:
Sorry about that. I forgot to add the check.